src/main.c

```diff
@@ -1,8 +1,120 @@
+#include <iostream>
+#include <string>
+#include <ctime>
+#include <vector>
+#include <sstream>
+#include <iomanip>
+#include <string.h>
+#include <windows.h>
+
+/********************Functions start**********************/
+static const std::string base64_chars =

```

Comment:
There is no `std::string` in C. It should be translated into C-style strings:

- Declare C strings using char arrays or dynamically allocated memory (`malloc`).
- Initialize strings with double quotes (e.g., `char str[] = "Hello";`).
- Replace C++ string operations:
  - Use `strcpy` for copying strings.
  - Use `strcat` for concatenation.
  - Use `strlen` for getting string length.

- Manage memory manually when using dynamic allocation (don't forget `free`).
- Ensure buffers are large enough, avoid overflows, and null-terminate strings properly.